taskvine/src/manager/vine_manager.c

```diff
@@ -2655,22 +2656,31 @@ static int vine_manager_transfer_capacity_available(struct vine_manager *q, stru
 		int found_match = 0;
 		
 		/* If not, then search for an available peer to provide it. */
-		/* Provide a substitute file object to describe the peer. */
 		
-		HASH_TABLE_ITERATE(q->worker_table, id, peer){
-			if((remote_info = hash_table_lookup(peer->current_files, f->cached_name)) && remote_info->in_cache) {
-				char *peer_source =  string_format("worker://%s:%d/%s", peer->transfer_addr, peer->transfer_port, f->cached_name);
-				if(vine_current_transfers_source_in_use(q, peer_source) < VINE_WORKER_SOURCE_MAX_TRANSFERS) {	
-					vine_file_delete(f->substitute);
-					f->substitute = vine_file_substitute_url(f,peer_source);
-					free(peer_source);
-					found_match = 1;
-					break;
-				} else {
-					free(peer_source);
-				}
+		/* If the file is a mini task specification, check its file dependencies for availability */
+		/* Do not peer transfer the mini task itself */
+		if(f->type==VINE_MINI_TASK) {
```

Comment:
I think this makes sense: we shouldn't peer-transfer the outputs of mini tasks.
But the control flow is confusing, each time I read it is a head scratcher.
Write it in the positive sense like this:
```
if(transfer transfer capacity available) {
  // do something
} else {
  // do something else
}
```

And if one of those is empty and falls through, then put int a big ol comment about what it is supposed to do.

Reply:
Yes the scheduling order did become a bit confusing. The purpose here was to maintain the priority of scheduling peer transfers. However we need to make sure we don't peer transfer mini-tasks due to complications later down the line (transferring mini task input files is fine).